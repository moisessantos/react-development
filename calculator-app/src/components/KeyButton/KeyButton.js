import React, { memo } from 'react';
import PropTypes from 'prop-types';

const KeyButton = ({ handleClick, text, ...otherProps }) => <button type="button" name={text} onClick={e => handleClick(e.target.name)} {...otherProps}>{text}</button>;

KeyButton.propTypes = {
  handleClick: PropTypes.func.isRequired,
  text: PropTypes.string.isRequired,
};

export default memo(KeyButton);
