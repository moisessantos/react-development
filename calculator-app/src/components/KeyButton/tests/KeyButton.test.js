import React from 'react';
import { shallow } from 'enzyme';
import KeyButton from '../';

describe('KeyButton tests', () => {
    it('should render as expected', () => {
        const wrapper = shallow(<KeyButton text='5' handleClick={ jest.fn() } />)
        expect(wrapper).toMatchSnapshot();
    });

    it('should call handleClick on button click', () => {
        const mock = jest.fn();
        const event = { target: { name: '5' } };
        const wrapper = shallow(<KeyButton text='5' handleClick={ mock } />);
        wrapper.find('button').simulate('click', event);

        expect(mock).toBeCalledTimes(1);
        expect(mock).toBeCalledWith(event.target.name);
    });
})