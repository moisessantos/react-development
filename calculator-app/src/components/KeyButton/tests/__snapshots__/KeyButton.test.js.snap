// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KeyButton tests should render as expected 1`] = `
<button
  name="5"
  onClick={[Function]}
  type="button"
>
  5
</button>
`;
